### PR TITLE
Rich text to job description draft

### PR DESCRIPTION
**Pull request by DanielMichelUno**
Draft pull request

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+*Please complete the following information*
```
Comment (geoperez) [inline .github/PULL_REQUEST_TEMPLATE.md]: @JCorrea13 this Repo is public, remove this template.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+name: Node CI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: [ubuntu-latest]
+    
+    steps:
+      - uses: actions/checkout@v1
```
Comment (geoperez) [inline .github/workflows/build.yml]: This is a really old Workflow, use the same as uno-js.

File: .github/workflows/lint-pr.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+name: Lint PR
+
+on: [pull_request]
+
+jobs:
+  eslint:
+    runs-on: ubuntu-latest
```
Comment (geoperez) [inline .github/workflows/lint-pr.yml:7]: Same

File: .npmrc [hunk 4]
```diff
@@ -0,0 +1 @@
+registry=https://npm.pkg.github.com/unosquare
```
Comment (geoperez) [inline .npmrc]: Public repo, we don't need this.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,139 @@
+{
+  "name": "@unosquare/oneonone-form",
```
Comment (geoperez) [inline package.json]: Rename the package to match the repo.

File: src/interfaces/IRichTexEditorProps.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+//import { EditorState } from 'draft-js';
+
+export interface IRichTextEditorProps {
```
Comment (geoperez) [inline src/interfaces/IRichTexEditorProps.ts]: Avoid adding prefix I to interfaces.

**Review by geoperez — CHANGES_REQUESTED** [review]

File: src/components/index.tsx [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+};
+const _convertToMark = (editor: EditorState) => {
+    const content = editor.getCurrentContent();
+    const raw = convertToRaw(content);
+    const mark = draftToMarkdown(raw, toMarkdownOptions);
+    return mark;
+};
+
+export type RichTextEditorRef = {
+    updateUI: (newMarkdown: string) => void;
```
Comment (JCorrea13) [inline src/components/index.tsx]: improve method name, maybe setContent? setText?

File: src/components/index.tsx [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+    const raw = markdownToDraft(markdown, fromMarkdownOptions);
+    const content = convertFromRaw(raw);
+    return EditorState.createWithContent(content);
+};
+const _convertToMark = (editor: EditorState) => {
+    const content = editor.getCurrentContent();
+    const raw = convertToRaw(content);
+    const mark = draftToMarkdown(raw, toMarkdownOptions);
+    return mark;
+};
```
Comment (JCorrea13) [inline src/components/index.tsx]: we can move this to a utils file

File: src/components/index.tsx [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+    blockStyles: {
+        ins_open: ['UNDERLINE'],
+    },
+    remarkableOptions: {
+        enable: {
+            inline: ['ins'],
+        },
+    },
+    preserveNewlines: true,
+};
```
Comment (JCorrea13) [inline src/components/index.tsx]: we can move it to its own file

File: src/components/index.tsx [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+        };
+
+        const applyBlockType = (event: any, style: string) => {
+            event.preventDefault();
+            handleChange(RichUtils.toggleBlockType(editorState, style));
+        };
+
+        const handleChange = (newState: EditorState) => {
+            setEditor(newState);
+        };
```
Comment (JCorrea13) [inline src/components/index.tsx]: do we need this function? or we can call setEditor instead and delete handleChange?

**Review by JCorrea13 — CHANGES_REQUESTED** [review]

**Review by MarcoLPR — APPROVED** [review]

File: src/components/index.tsx (hunk 9 above)
Comment (DanielMichelUno) [reply src/components/index.tsx]: How would you name that file?

File: .eslintrc.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
+module.exports = {
+  parser: "@typescript-eslint/parser", // Specifies the ESLint parser
+  extends: [
```
Comment (geoperez) [inline .eslintrc.js:3]: I update the extends, check wp common

File: .ncurc.json [hunk 12]
```diff
@@ -0,0 +1,7 @@
+{
+    "upgrade": true,
+    "reject": [
+        "react",
+        "react-dom"
```
Comment (geoperez) [inline .ncurc.json]: Also, ignore react and react-dom types.

File: package.json [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+    "typescript": "^4.1.2",
+    "webpack": "^5.10.0",
+    "webpack-cli": "^4.5.0",
+    "webpack-dev-server": "^3.11.0",
+    "react": "16.14.0",
+    "react-dom": "16.14.0",
+    "react-select": "3.1.0"
+  },
+  "peerDependencies": {
+    "react": "16.14.0",
```
Comment (geoperez) [inline package.json]: Change to react 16.8.5 pls

File: package.json [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+    "react-dom": "16.14.0",
+    "react-select": "3.1.0"
```
Comment (geoperez) [inline package.json]: Do you really need the dep react select?

File: package.json [hunk 15]
```diff
@@ -0,0 +1,137 @@
...
+  },
+  "dependencies": {
+    "@fluentui/react": "^7.153.4",
+    "@microsoft/applicationinsights-react-js": "^3.0.5",
```
Comment (geoperez) [inline package.json]: Remove app insights, clean up deps. I see many that you don't need.

File: package.json [hunk 16]
```diff
@@ -0,0 +1,137 @@
...
+    ],
+    "coverageThreshold": {
+      "global": {
+        "branches": 50,
+        "functions": 50,
+        "lines": 80,
+        "statements": -230
+      }
+    },
+    "moduleNameMapper": {
```
Comment (geoperez) [inline package.json:103]: These mappers are for Fluent, if this component is not fluent, remove it.

File: sample/src/main.tsx [hunk 17]
```diff
@@ -0,0 +1,31 @@
+import React, { useState, useRef } from 'react';
+import { UnoReactDraftjs, RichTextEditorRef } from '../../../uno-react-draftjs/src/index';
```
Comment (geoperez) [inline sample/src/main.tsx]: WTF?

**Review by geoperez — CHANGES_REQUESTED** [review]

File: sample/src/main.tsx (hunk 17 above)
Comment (DanielMichelUno) [reply sample/src/main.tsx]: What would be the issue?

File: package.json (hunk 16 above)
Comment (DanielMichelUno) [reply package.json:103]: The component uses fluent styles.

Comment (geoperez) [reply package.json:103]: Aright then.

**Review by JCorrea13 — APPROVED** [review]

**Review by geoperez — APPROVED** [review]

**Pull request merged**